src/backend/joanie/tests/core/models/order/test_schedule.py

```diff
@@ -0,0 +1,694 @@
+            order = factories.OrderFactory()
+
+        self.assertEqual(
+            order.retraction_date(),
+            datetime(2024, 1, 2, 0, 0, tzinfo=ZoneInfo("UTC")),
+        )
+
+    def test_models_order_schedule_get_installments_percentages(self):
+        """
+        Check that the correct payment limits are returned for different amounts
+        """
+        order = factories.OrderFactory(product__price=3)
+        self.assertEqual(order.get_installments_percentages(), (30, 70))
+        order = factories.OrderFactory(product__price=5)
+        self.assertEqual(order.get_installments_percentages(), (30, 70))
+
+        order = factories.OrderFactory(product__price=7)
+        self.assertEqual(order.get_installments_percentages(), (30, 45, 45))
+        order = factories.OrderFactory(product__price=10)
+        self.assertEqual(order.get_installments_percentages(), (30, 45, 45))
+
+        order = factories.OrderFactory(product__price=80)
+        self.assertEqual(order.get_installments_percentages(), (20, 30, 30, 20))
+        order = factories.OrderFactory(product__price=100)
+        self.assertEqual(order.get_installments_percentages(), (20, 30, 30, 20))
+        order = factories.OrderFactory(product__price=150)
+        self.assertEqual(order.get_installments_percentages(), (20, 30, 30, 20))
+
+    def test_model_order_schedule_calculate_due_dates(self):
+        """
+        Check that the due dates are correctly calculated
+        """
+        start_date = datetime(2024, 1, 1, 0, tzinfo=ZoneInfo("UTC"))
+        end_date = datetime(2024, 3, 20, 8, tzinfo=ZoneInfo("UTC"))
+
+        with patch(
+            "joanie.core.models.products.Order.get_schedule_dates"
+        ) as mock_get_schedule_dates:
+            mock_get_schedule_dates.return_value = start_date, end_date
+            order = factories.OrderFactory(product__price=3)
+            due_dates = order.calculate_due_dates(2)
+
+        self.assertEqual(
+            due_dates,
+            [
+                datetime(2024, 1, 1, 0, 0, tzinfo=ZoneInfo("UTC")),
+                datetime(2024, 2, 1, 0, 0, tzinfo=ZoneInfo("UTC")),
+            ],
+        )
+
+    def test_model_order_schedule_calculate_due_dates_end_date(self):
+        """
+        Check that the due dates are correctly calculated
+        when the end date is before the second part
+        """
+        start_date = datetime(2024, 1, 1, 0, tzinfo=ZoneInfo("UTC"))
+        end_date = datetime(2024, 1, 20, 8, tzinfo=ZoneInfo("UTC"))
+
+        with patch(
+            "joanie.core.models.products.Order.get_schedule_dates"
+        ) as mock_get_schedule_dates:
+            mock_get_schedule_dates.return_value = start_date, end_date
+            order = factories.OrderFactory(product__price=3)
+            due_dates = order.calculate_due_dates(2)
+
+        self.assertEqual(
+            due_dates,
+            [
+                datetime(2024, 1, 1, 0, 0, tzinfo=ZoneInfo("UTC")),
```

Comment:
Shouldn't we have a second due date that should be `2024-01-20` ?

Reply:
I assumed that having 2 payments on the same month was strange.

Reply:
I don't know if we should think with a salary period. If you have 20 days between two payments and you inform them of this IMO it's good.
Moreover, it is forbidden to debit more than 30% on the first payment, so your test prove here that it's not the good solution I think.

Reply:
Oh, ok, I'll change that.